cms/signals/slug_update_fix.py

```diff
@@ -0,0 +1,32 @@
+"""
+This is a temporary fix that should be replaced with an adequate bug fix to the Page logic.
+"""
+
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+from fieldsignals import post_save_changed
+
+from cms.models import Title, Page
+
+
+@receiver(post_save, sender=Page)
+def fix_page_slugs(sender, instance, **kwargs):
+    _update_child_pages(instance)
+
+
+@receiver(post_save_changed, sender=Title, fields=['slug', 'path'])
+def fix_title_path(sender, instance, **kwargs):
```

Comment:
We do have signals already created, we should just bind to those. V4 reworks the signal in a better fashion.